app/dashboard/views.py

```diff
@@ -1,32 +1,32 @@
 from rest_framework import viewsets
-from rest_framework.authentication import TokenAuthentication, \
-    SessionAuthentication
 from rest_framework.permissions import IsAuthenticated
-
+from squac.mixins import SetUserMixin, PermissionsMixin
+from squac.permissions import IsAdminOwnerOrPublicReadOnly
 from .models import Dashboard, WidgetType, Widget, StatType
 from dashboard import serializers
 
 
-class BaseDashboardViewSet(viewsets.ModelViewSet):
-    '''base class for dashboard viewsets'''
-
-    authentication_classes = (TokenAuthentication, SessionAuthentication)
-    permission_classes = (IsAuthenticated, )
-
-    # all models require an auth user, set on create
-    def perform_create(self, serializer):
-        serializer.save(user=self.request.user)
+class BaseDashboardViewSet(SetUserMixin, PermissionsMixin,
+                           viewsets.ModelViewSet):
+    pass
 
 
 class DashboardViewSet(BaseDashboardViewSet):
     serializer_class = serializers.DashboardSerializer
+    permission_classes = (
+        IsAuthenticated, IsAdminOwnerOrPublicReadOnly)
     queryset = Dashboard.objects.all()
 
     def get_serializer_class(self):
         if self.action == 'retrieve':
             return serializers.DashboardDetailSerializer
         return self.serializer_class
 
+    def get_queryset(self):
+        queryset = Dashboard.objects.filter(user=self.request.user) | \
+            Dashboard.objects.filter(is_public=True)
```

Comment:
What if the user making the request has permission to view all dashboards (Admin?)